src/briefcase/commands/base.py

```diff
@@ -117,94 +108,54 @@ class BaseCommand(ABC):
-        # indicates we're under test conditions. If there's no data path
-        # provided, look for a BRIEFCASE_HOME environment variable. If that
-        # isn't defined, use a platform-specific default data path.
-        if data_path is None:
-            try:
-                briefcase_home = os.environ["BRIEFCASE_HOME"]
-                data_path = Path(briefcase_home).resolve()
-                # Path("") converts to ".", so check for that edge case.
-                if briefcase_home == "" or not data_path.exists():
-                    raise BriefcaseCommandError(
-                        "The path specified by BRIEFCASE_HOME does not exist."
-                    )
-            except KeyError:
-                if self.host_os == "Darwin":
-                    # macOS uses a bundle name, rather than just the app name
-                    app_name = "org.beeware.briefcase"
-                else:
-                    app_name = "briefcase"
-
-                data_path = PlatformDirs(
-                    appname=app_name,
-                    appauthor="BeeWare",
-                ).user_cache_path
-
-        if " " in os.fsdecode(data_path):
-            raise BriefcaseCommandError(
-                f"""
-The location Briefcase will use to store tools and support files:
-
-    {data_path}
-
-contains spaces. This will cause problems with some tools, preventing
-you from building and packaging applications.
-
-You can set the environment variable BRIEFCASE_HOME to specify
-a custom location for Briefcase's tools.
-
-"""
-            )
-
-        self.data_path = Path(data_path)
-
-        self.tools_path = self.data_path / "tools"
+        self.tools = tools or ToolCache(
+            logger=logger,
+            console=console,
+            base_path=self.data_path / "tools",
+        )
 
         self.global_config = None
-        self.apps = {} if apps is None else apps
         self._path_index = {}
 
-        # External service APIs.
-        # These are abstracted to enable testing without patching.
-        self.cookiecutter = cookiecutter
-        self.requests = requests
-        self.input = Console(enabled=input_enabled)
-        self.os = os
-        self.sys = sys
-        self.stdlib_platform = platform
-        self.shutil = shutil
-        self.subprocess = Subprocess(self)
+    @property
+    def logger(self):
+        return self.tools.logger
+
+    @logger.setter
+    def logger(self, val):
+        raise RuntimeError("'logger' is a readonly property for Commands")
 
-        # The internal Briefcase integrations API.
-        self.integrations = integrations
+    @property
+    def input(self):
+        return self.tools.input
 
-        # Initialize default logger (replaced when options are parsed).
-        self.logger = Log()
-        self.save_log = False
+    @input.setter
+    def input(self, val):
+        raise RuntimeError("'input' is a readonly property for Commands")
```

Comment:
I wasn't sure if you _literally_ meant these should be read-only....but they are.

Reply:
TIL :) removed setters